src/components/screens/timebank-view-all-screen.tsx

```diff
@@ -0,0 +1,290 @@
+            padding: theme.spacing(1)
+          }}
+        >
+          {`${sectionName}: ${getHoursAndMinutes(selectedData.value as number)}`}
+        </Typography>
+      </Box>
+    );
+  };
+
+  /**
+   * Render piechart component
+   *
+   * @param personWithTotalTime person with total time object
+   * @param legend legend boolean
+   */
+  const renderPieChart = (personWithTotalTime: PersonWithTotalTime, legend: boolean) => {
+    const { person, personTotalTime } = personWithTotalTime;
+
+    if (!person || !personTotalTime) return null;
+
+    const workTimeData: WorkTimeTotalData[] = [
+      { name: WorkTimeCategory.BILLABLE_PROJECT, balance: personTotalTime.billableProjectTime },
+      {
+        name: WorkTimeCategory.NON_BILLABLE_PROJECT,
+        balance: personTotalTime.nonBillableProjectTime
+      },
+      { name: WorkTimeCategory.INTERNAL, balance: personTotalTime.internalTime }
+    ];
+
+    return (
+      <ResponsiveContainer width={"75%"} height={150}>
+        <PieChart>
+          <Pie
+            cx="50%"
+            cy="50%"
+            dataKey="balance"
+            data={workTimeData}
+            label={(props) => getHoursAndMinutes(props.value)}
+          >
+            {workTimeData.map((_entry, index) => (
+              <Cell key={`cell-${index}`} fill={COLORS[index % COLORS.length]} />
+            ))}
+          </Pie>
+          {legend ? <Legend wrapperStyle={{ position: "relative" }} /> : null}
+          <Tooltip content={renderCustomizedTooltip} />
+        </PieChart>
+      </ResponsiveContainer>
+    );
+  };
+
+  /**
+   * Populate one person's total work data
+   *
+   * @param person person with total time data
+   */
+  const populatePersonTotalTimeData = async (
+    person: PersonWithTotalTime
+  ): Promise<PersonWithTotalTime> => {
+    let totalTime: PersonTotalTime[] = [];
+
+    try {
+      totalTime = await personsApi.listPersonTotalTime({
+        personId: person.person.id,
+        timespan: Timespan.ALL_TIME
+      });
+    } catch (error) {
+      setError(`${error} ${strings.error.totalTimeFetch}`);
+    }
+
+    return {
+      ...person,
+      personTotalTime: totalTime[0]
+    };
+  };
+
+  /**
+   * Fetches the person data & person total time data
+   */
+  const fetchPersonsAndPersonsTotalTime = async () => {
+    setLoading(true);
+
+    try {
+      let fetchedPersons: Person[] = persons;
+
+      if (!fetchedPersons.length) {
+        fetchedPersons = await personsApi.listPersons({
+          active: true
+        });
+        setPersons(fetchedPersons);
+      }
+
+      const personsTimeTotals = await Promise.all(
+        persons
+          .map((person) => ({
+            person: person
+          }))
+          .map(populatePersonTotalTimeData)
+      );
+
+      setPersonsWithTotalTime(personsTimeTotals);
+      setDisplayedPersonsTotalTime(personsTimeTotals);
+    } catch (error) {
+      setError(`${error} ${strings.error.fetchFailedGeneral}`);
```

Comment:
More specific error messages help to identify where problems are coming from. In this case something more specific like "error fetching persons" should be fine as the populatePersonTotalTimeData has its own catch,

Reply:
Made a new localized string for this catch and applied it